packages/dcos-launch/extra/launch/config.py

```diff
@@ -1,15 +1,37 @@
+import atexit
 import copy
 import os
 
 import yaml
 
-import dcos_installer.config
 import gen
 import launch.util
 import ssh.validate
 import test_util.aws
 from gen.internals import resolve_configuration, Scope, Source, Target, validate_one_of
-from pkgpanda.util import load_string, load_yaml, YamlParseError
+from pkgpanda.util import load_string, load_yaml, YamlParseError  # n
+
+# dcos_installer.config will directly import gen.build_deploy.util
+# which expects to find the image commit in the environment or in
+# a directory-local git tree **at import time**. Therefore, the
+# environment variable DCOS_IMAGE_COMMIT must be set here.
+if 'DCOS_IMAGE_COMMIT' not in os.environ:
+    os.environ['DCOS_IMAGE_COMMIT'] = 'deadbeef'
```

Comment:
Sure. I didn't feel too strongly, but I like deadbeef because its obviously wrong on visual inspection. However deadbeef could fool a small sanity test, so its probably safer to use `''`